frontend/src/hooks/useError.ts

```diff
@@ -1,21 +1,51 @@
-import { useEffect, useContext } from 'react'
-import { AxiosError } from 'axios'
-import ErrorContext from 'contexts/ErrorContext'
+import { useEffect, useContext } from 'react';
+import { AxiosError } from 'axios';
+import ErrorContext from 'contexts/ErrorContext';
 
-export default function useError ({ error, prefix }: {
-  error: AxiosError | undefined,
-  prefix?: string,
+export default function useError({
+  error,
+  prefix,
+}: {
+  error: AxiosError | undefined;
+  prefix?: string;
 }) {
-  const { updateErrorContext } = useContext(ErrorContext)
+  const { updateErrorContext } = useContext(ErrorContext);
   useEffect(() => {
     if (error) {
-      error.message = error.message || 'An unknown error occurred.'
-      if (error.response?.data?.message) {
-        error.message = error.response?.data?.message
+      let message = error.message || 'An unknown error occurred.';
+      try {
+        // Decode error response
+        if (error.response && error.response.data) {
+          const data = error.response.data;
+          if (data.error) {
+            if (data.error.constructor === Array) {
+              const errors: string[] = data.error || [];
+              message = errors.join('/');
+            } else {
+              message = `${error}`;
+            }
+          } else {
+            const dataMessage = data.message;
+            if (dataMessage) {
+              if (dataMessage.constructor === Array) {
+                message = dataMessage.join('/');
+              } else {
+                message = dataMessage;
+              }
+            }
+          }
+        }
+
+        const errorMessage = `${message} (${error.response?.statusText})`;
+        const finalMessage = prefix
+          ? `${prefix} Error: ${errorMessage}`
+          : `Unable to perform operation. Error: ${errorMessage}`;
+        updateErrorContext(finalMessage);
+      } catch (excp) {
```

Comment:
Not sure I like a try/catch around pure object manipulation code. I would assume any error is a missing case in the if/elses.

Reply:
@will-freshworks , I put this because in case axios error response structure changes .....